src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -2500,6 +2500,9 @@ FIXTURE_TEST(read_write_truncate, storage_test_fixture) {
       [&] { return cnt > max; },
       [&log, &cnt] {
           auto offset = log->offsets();
+          if (offset.dirty_offset <= model::offset(0)) {
```

Comment:
IIUC the issue is not that its less than zero, it's that it is more like numeric_limits<int64_t>::min() or something?

Reply:
Replaced with comparison with `model::offset{}` instead.